client/src/main/java/io/pravega/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -0,0 +1,367 @@
+
+    /**
+     * List of codec types used for encoding in the group. 
+     * 
+     * @param groupId Id for the group. 
+     * @return List of codec types used for encoding in the group. 
+     * @throws ResourceNotFoundException if group is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    List<String> getCodecTypes(String groupId) throws ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Add new codec type to be used in encoding in the group. Adding a new codectype is backward incompatible. 
+     * Make sure all readers are upgraded to use the new codec before any writers use the codec to encode the data. 
+     * 
+     * @param groupId Id for the group. 
+     * @param codecType codec type.
+     * @throws ResourceNotFoundException if group is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    void addCodecType(String groupId, String codecType) throws ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Gets complete schema evolution history of the group with schemas, versions, compatibility policy and  
+     * time when the schema was added to the group. 
+     * The order in the list matches the order in which schemas were evolved within the group. 
+     * This call is atomic and will get a consistent view at the time when the request is processed on the service. 
+     * So all schemas that were added before this call are returned and all schemas that were deleted before this call
+     * are excluded. 
+     *
+     * @param groupId Id for the group.
+     * @return Ordered list of schemas with versions and compatibility policy for all schemas in the group. 
+     * @throws ResourceNotFoundException if group is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    List<GroupHistoryRecord> getGroupHistory(String groupId) throws ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Finds all groups and corresponding version info for the groups where the supplied schema has been registered.
+     * It is important to note that the same schema type could be part of multiple group, however in each group it 
+     * may have gone through a separate evolution. Invocation of this method lists all groups where the specific schema 
+     * (type, format and binary) is used along with versions that identifies this schema in those groups. 
+     * The user defined {@link SchemaInfo#properties} is not used for comparison. 
+     * 
+     * @param schemaInfo Schema info to find references for. 
+     * @return Map of group Id to versionInfo identifier for the schema in that group. 
+     * @throws ResourceNotFoundException if schema is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    Map<String, VersionInfo> getSchemaReferences(SchemaInfo schemaInfo) throws ResourceNotFoundException, UnauthorizedException;
```

Comment:
I think there is a use for this API, but perhaps it should not be in 1.0.
This has the same equality check problem mentioned above. If more fields are on SchemaInfo we need to decide how to deal with them. 
If the backend is going to scale I would think this API would need to be eventually consistent. I am not sure how the schemaInfo index could be updated atomically with respect to the addition of the schema to the group. 
Given these distinctions I think it makes sense for this to be part of a dynamic querying API that is separate from the main client API. We could easily add a lot of features to such an api, including full indexing. However for now I think we should focus on the APIs needed for the Data path.

Reply:
As explained in the previous comment, the schema equivalence check is essential and used in 3 places. So this api relies on the exact same check. 

This could be a fairly useful api for data discovery -- find all streams where such data is being written.